src/main/java/eu/dissco/core/digitalmediaobjectprocessor/service/ProcessingService.java

```diff
@@ -66,13 +70,29 @@ public List<DigitalMediaObjectRecord> handleMessage(List<DigitalMediaObjectTrans
     return results;
   }
 
-  private Set<DigitalMediaObjectTransferEvent> removeDuplicatesInBatch(
-      List<DigitalMediaObjectTransferEvent> events) {
-    var uniqueSet = new HashSet<DigitalMediaObjectTransferEvent>();
+  private void checkIfDigitalSpecimenIdExists(List<DigitalMediaObjectEvent> events)
+      throws DigitalSpecimenNotFoundException {
+    var digitalSpecimenIds = events.stream().map(DigitalMediaObjectEvent::digitalMediaObject)
```

Comment:
Switched it. However stream work a bit different they won't iterate multiple times over the array. When the terminal operator is called it will run once over the stream with all operations that are called. So both the one map and two maps won't have a performance impact.We can just do what we think looks best.